slack/src/main/java/com/hazelcast/jet/contrib/slack/SlackSinks.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    private static HttpRequestRetryHandler httpRetryHandler() {
+        return (exception, executionCount, context) -> {
+            if (executionCount >= RETRY_COUNT) {
+                // Do not retry if over max retry count
+                return false;
+            }
+            if (exception instanceof UnknownHostException) {
+                // Unknown host
+                return false;
+            }
+            if (exception instanceof ConnectTimeoutException) {
+                // Connection refused
+                return true;
+            }
+            if (exception instanceof HttpHostConnectException) {
+                // connection exception
+                return true;
+            }
+            if (exception instanceof IOException) {
+                // Interrupted Io exceptions
+                return true;
+            }
+            HttpClientContext clientContext = HttpClientContext.adapt(context);
+            HttpRequest request = clientContext.getRequest();
+            boolean idempotent = !(request instanceof HttpEntityEnclosingRequest);
+            if (idempotent) {
+                // Retry if the request is considered idempotent
+                return true;
+            }
+            return false;
+        };
+    }
+
+    private static final class SlackContext {
+
+        private final SupplierEx<CloseableHttpClient> httpClientSupplierEx;
+        private final CloseableHttpClient closeableHttpClient;
+        private final String accessToken;
+        private final String channel;
+
+        private SlackContext(SupplierEx<CloseableHttpClient> bulkRequestSupplier, String accessToken, String channel) {
+            this.httpClientSupplierEx = bulkRequestSupplier;
+            this.closeableHttpClient = bulkRequestSupplier.get();
+            this.accessToken = accessToken;
+            this.channel = channel;
+        }
+
+        private String receiveFn(String message) {
+            HttpPost request =  new HttpPost(URL);
+            CloseableHttpResponse response = null;
+            String result = "";
+            try {
+                // add request headers
+                request.addHeader("Authorization", String.format("Bearer %s", accessToken));
+                List<NameValuePair> urlParameters = new ArrayList<>();
+                urlParameters.add(new BasicNameValuePair("channel",  channel));
+                urlParameters.add(new BasicNameValuePair("text", message));
+                request.setEntity(new UrlEncodedFormEntity(urlParameters));
+                response = (CloseableHttpResponse) closeableHttpClient.execute(request);
+                result = EntityUtils.toString(response.getEntity());
+            } catch (UnsupportedEncodingException var) {
+                throw ExceptionUtil.rethrow(var);
+            } catch (ClientProtocolException var) {
+                throw ExceptionUtil.rethrow(var);
+            } catch (IOException var) {
```

Comment:
You can combine these three:
```java
} catch (UnsupportedEncodingException | ClientProtocolException | IOException var) {
```

Reply:
@viliam-durina `UnsupportedEncodingException  & ClientProtocolException ` are chile classes of `IOException `. Updated the changes here.